config/files/server/usr/etc/ssh/sshd_config.d/99-hardening.conf

```diff
@@ -20,4 +26,32 @@ TCPKeepAlive no
 ClientAliveCountMax 2
 
 # Forbid sshing as root
-PermitRootLogin no
\ No newline at end of file
+PermitRootLogin no
+
+# Enable Pubkey Authentification
+PubkeyAuthentication yes
+
+# Block empty passwords
+PermitEmptyPasswords no
+
+# Allow only 256bit AES encryption
+Ciphers aes256-ctr
+
+# Allow only 512bit SHA-2 HMAC
+MACs hmac-sha2-512
```

Comment:
why is this needed?

Reply:
the Message Authentication Codes are used by ssh for integrity verification of transmitted data, this should protect against some brute force attacks

https://security.stackexchange.com/a/117104

Reply:
okay I changed those again. Added the cyphers from the post, and allowed the standard ones too for following the last mentioned reason.

Reply:
I am actually very unsure if we should block everything but the most secure protocols, with the possibility of breaking clients (add a note that they should use a different client) or keep those protocols.